File: DataAccessLayer/Entities/Album.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DataAccessLayer.Entities
{
    public class Album: BaseEntity
    {
        [Required, StringLength(100)]
        public string Title { get; set; }

        [StringLength(300)]
        public string Description { get; set; }

        public int NumberOfPictures { get; set; }

        [Required]
        [DatabaseGenerated(DatabaseGeneratedOption.None)]
        public int UserId { get; set; }

        [Required]
        public DateTime CreationDate { get; set; }
        
        
        public User User { get; set; }
        public ICollection<AlbumPhoto> AlbumPhotos { get; set; }
    }
}

File: UI-Angular/src/app/models/album.ts
export class Album{
    id!:number;
    title: string;
    description: string;
    numberOfPictures: number;
    userId: number;
    creationDate: Date;
    albumPhotoIds: Array<number>;

    constructor(title:string, description:string, numberOfPictures:number
        , userId: number, creationDate:Date, albumPhotoIds: Array<number>){

            this.title = title;
            this.description = description;
            this.numberOfPictures = numberOfPictures;
            this.userId = userId;
            this.creationDate = creationDate;
            this.albumPhotoIds = albumPhotoIds;
        }
}